src/scripts/sync_synonym.py

```diff
@@ -175,28 +229,67 @@ def sync_synonyms(
     mondo_exclusions_df = mondo_exclusions_df.drop(columns=['hasDbXref_xref', 'source_xref'])
     mondo_exclusions_df['synonym_lower'] = mondo_exclusions_df['synonym'].str.lower()
 
     # Query synonyms: source
+    # todo: ideally refactor to use only SPARQL instead of combining SPARQL & OAK results
+    #  - likely quick and easy but there's a chance there's more to it
     source_df: pd.DataFrame = _query_synonyms(mappings_df['source_id'].tolist(), source_db)\
         .rename(columns={'curie': 'source_id'})
     source_df['synonym_lower'] = source_df['synonym'].str.lower()
     source_df['source_label'] = source_df['source_id'].map(source_labels)
+    source_df.drop_duplicates(inplace=True)
+    # todo: some of these will be CURIE, e.g. MONDO:GENERATED. others URI, e.g.
+    #  http://purl.obolibrary.org/obo/OMO_0003012. Need to find a way to standardize. Probably best to do URI because
+    #  harder to account for all possible namespaces.
+    # - get synonym_types: declared by the source
+    source_types_df: pd.DataFrame = _read_sparql_output_tsv(onto_synonyms_path).rename(columns={'cls_id': 'source_id'})
+    # -- remove synonym xref provenance
+    # todo: it may be useful in the future to keep/use this data
+    del source_types_df['dbXref']  # leaves: source_id, synonym_scope, synonym, synonym_type
+    # -- URI --> CURIE: source_id
+    source_types_df['source_id'] = source_types_df['source_id'].apply(lambda x: conv.compress(x))
+    # -- filter non-source-owned terms (i.e. imported from other ontologies, e.g. ChEBI)
+    source_types_df = source_types_df[~source_types_df['source_id'].isna()]
+    # -- CURIE --> URI: synonym_type
+    # todo: Ideally use a cuires prefix_map flexible enough to handle Mondo & source(s)
+    source_types_df['synonym_type'] = source_types_df['synonym_type'].apply(
+        lambda x: x.replace('MONDO:', 'http://purl.obolibrary.org/obo/mondo#'))
+    uri_prefixes: Set[str] = set([x.split(':')[0] for x in source_types_df['synonym_type']]).difference({'', 'http'})
+    if uri_prefixes:
+        raise RuntimeError('Error: CURIE(s) detected in URI-typed field "synonym_type": ' + ', '.join(uri_prefixes))
+    # -- remove language codes
+    source_types_df['synonym'] = source_types_df['synonym'].apply(_remove_language_code)
+    source_types_df.drop_duplicates(inplace=True)
+    # -- filter out rows with no types; can cause duplicate rows (other rows were probably from xref axioms)
+    source_types_df = source_types_df[source_types_df['synonym_type'] != '']
+    # -- multiple synonym_types: QC
+    # TODO: probably need to support multiple types
+    source_dupe_types_df = source_types_df[source_types_df.duplicated(
+        subset=['source_id', 'synonym', 'synonym_scope'], keep=False)]
+    if len(source_dupe_types_df) > 0:
+        raise RuntimeError('Error: Unhandled multiple synonym_types detected in Mondo')
```

Comment:
## TODO: Support reading 2+ `synonym_type` (source & Mondo)
We can, and do, have instances of multiple `synonym_type` in the ROBOT template.

However, one thing I didn't have time to do yet was to handle edge cases of multiple synonym types are when querying the axioms from Mondo or sources.

For now, I've set up an error to alert when this happens. I got no error, so there are no current instances. However, there will be in the future. That's why I've marked this with an uppercase `TODO`; I will handle it soon.

Reply:
Comment out this section for now until we can discuss how this should be handled.

Reply:
Discussed and we will prioritize this soon; likely next week.